WcaOnRails/app/controllers/registrations_controller.rb

```diff
@@ -183,7 +183,12 @@ def update
       end
     else
       flash.now[:danger] = I18n.t('registrations.flash.failed')
-      render :edit
+      if params[:from_admin_view]
+        render :edit
+      else
+        self.register # Set instance variables needed by :register view
```

Comment:
Good call! I've made that change. I'm not a big fan of `before_action`s, and I'm not sure it would be the right thing to use in this `def update`, where oftentimes we'd end up computing this instance variables and then redirecting.

Good call on `process_payments`. I've also attempted to change it to use this new `set_registration_info!` method (yes, I made it a bang method =))